<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>forth32</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>851</width>
    <height>619</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>HuaweiFlasher</string>
  </property>
  <property name="windowIcon">
   <iconset resource="qhuaweiflash.qrc">
    <normaloff>:/icon.ico</normaloff>:/icon.ico</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>15</y>
      <width>171</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Прошивочный порт:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="PortSelector">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>10</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="partlist">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>100</y>
      <width>181</width>
      <height>261</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>70</y>
      <width>161</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Таблица разделов</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>424</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Идент.</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>454</y>
      <width>54</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Дата</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>480</y>
      <width>54</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Время</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>87</x>
      <y>515</y>
      <width>54</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Версия</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="Platform_input">
    <property name="geometry">
     <rect>
      <x>49</x>
      <y>420</y>
      <width>71</width>
      <height>27</height>
     </rect>
    </property>
    <property name="maxLength">
     <number>8</number>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="Date_input">
    <property name="geometry">
     <rect>
      <x>48</x>
      <y>450</y>
      <width>101</width>
      <height>27</height>
     </rect>
    </property>
    <property name="maxLength">
     <number>16</number>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="Time_input">
    <property name="geometry">
     <rect>
      <x>48</x>
      <y>478</y>
      <width>101</width>
      <height>27</height>
     </rect>
    </property>
    <property name="maxLength">
     <number>16</number>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="Version_input">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>540</y>
      <width>194</width>
      <height>27</height>
     </rect>
    </property>
    <property name="maxLength">
     <number>32</number>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>70</y>
      <width>191</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Содержимое раздела</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>50</y>
      <width>841</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>27</x>
      <y>383</y>
      <width>171</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Заголовок раздела</string>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>-3</x>
      <y>370</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>60</y>
      <width>20</width>
      <height>531</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QToolButton" name="RefreshPorts">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>13</y>
      <width>25</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>...</string>
    </property>
    <property name="iconSize">
     <size>
      <width>20</width>
      <height>20</height>
     </size>
    </property>
   </widget>
   <widget class="QToolButton" name="setdate">
    <property name="geometry">
     <rect>
      <x>156</x>
      <y>463</y>
      <width>54</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Тек.</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="pcode">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>420</y>
      <width>51</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>128</x>
      <y>424</y>
      <width>54</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Код</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>851</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="fileopen"/>
    <addaction name="fileappend"/>
    <addaction name="separator"/>
    <addaction name="filesave"/>
    <addaction name="separator"/>
    <addaction name="file_exit"/>
   </widget>
   <widget class="QMenu" name="menu_oper">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="title">
     <string>Операции</string>
    </property>
    <addaction name="Menu_Oper_flash"/>
    <addaction name="Menu_Oper_USBDload"/>
    <addaction name="Menu_Oper_Reboot"/>
   </widget>
   <widget class="QMenu" name="menu_part">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Раздел</string>
    </property>
    <addaction name="MoveUp"/>
    <addaction name="MoveDown"/>
    <addaction name="Delete"/>
    <addaction name="separator"/>
    <addaction name="part_edit_header"/>
    <addaction name="separator"/>
    <addaction name="part_store"/>
    <addaction name="part_extract"/>
    <addaction name="part_replace"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_part"/>
   <addaction name="menu_oper"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="fileopen">
   <property name="text">
    <string>Открыть</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="fileappend">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Добавить</string>
   </property>
  </action>
  <action name="part_store">
   <property name="text">
    <string>Извлечь с заголовком</string>
   </property>
  </action>
  <action name="part_extract">
   <property name="text">
    <string>Извлечь без заголовка</string>
   </property>
  </action>
  <action name="part_replace">
   <property name="text">
    <string>Заменить образ раздела</string>
   </property>
  </action>
  <action name="file_exit">
   <property name="text">
    <string>Выход</string>
   </property>
  </action>
  <action name="filesave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Сохранить полный образ прошивки</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="MoveUp">
   <property name="text">
    <string>Сдвинуть вверх</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Up</string>
   </property>
  </action>
  <action name="MoveDown">
   <property name="text">
    <string>Сдвинуть вниз</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Down</string>
   </property>
  </action>
  <action name="Delete">
   <property name="text">
    <string>Удалить</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Del</string>
   </property>
  </action>
  <action name="part_edit_header">
   <property name="text">
    <string>Редактировать заголовок</string>
   </property>
  </action>
  <action name="Menu_Oper_flash">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Прошивка</string>
   </property>
   <property name="shortcut">
    <string>Alt+B</string>
   </property>
  </action>
  <action name="Menu_Oper_USBDload">
   <property name="text">
    <string>Аварийная USB-загрузка</string>
   </property>
   <property name="shortcut">
    <string>Alt+U</string>
   </property>
  </action>
  <action name="Menu_Oper_Reboot">
   <property name="text">
    <string>Перезагрузка модема</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="qhuaweiflash.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>fileopen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>SelectFwFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>partlist</sender>
   <signal>itemActivated(QListWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>SelectPart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>288</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>307</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileappend</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>AppendFwFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>part_extract</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Menu_Part_Extract()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>part_store</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Menu_Part_Store()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>partlist</sender>
   <signal>itemClicked(QListWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>SelectPart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>95</x>
     <y>292</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>part_replace</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Menu_Part_Replace()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>file_exit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Terminate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filesave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>SaveFwFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Delete</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Menu_Part_Delete()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MoveUp</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Menu_Part_MoveUp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MoveDown</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Menu_Part_MoveDown()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>part_edit_header</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Menu_Part_EditHeader()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Platform_input</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>HeaderChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>473</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Date_input</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>HeaderChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>503</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Time_input</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>HeaderChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>533</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Version_input</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>HeaderChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>593</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>partlist</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>Disable_EditHeader()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>262</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Menu_Oper_flash</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Start_Flasher()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Menu_Oper_Reboot</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Reboot_modem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Menu_Oper_USBDload</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>usbdload()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RefreshPorts</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>find_ports()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>471</x>
     <y>52</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>52</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setdate</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>set_date()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>504</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>509</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pcode</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>HeaderChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>249</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>SelectFwFile()</slot>
  <slot>SelectPart()</slot>
  <slot>AppendFwFile()</slot>
  <slot>Menu_Part_Store()</slot>
  <slot>Menu_Part_Extract()</slot>
  <slot>Menu_Part_Replace()</slot>
  <slot>Terminate()</slot>
  <slot>SaveFwFile()</slot>
  <slot>Menu_Part_MoveUp()</slot>
  <slot>Menu_Part_MoveDown()</slot>
  <slot>Menu_Part_Delete()</slot>
  <slot>Menu_Part_EditHeader()</slot>
  <slot>Disable_EditHeader()</slot>
  <slot>HeaderChanged()</slot>
  <slot>Start_Flasher()</slot>
  <slot>Reboot_modem()</slot>
  <slot>usbdload()</slot>
  <slot>find_ports()</slot>
  <slot>set_date()</slot>
 </slots>
</ui>
